internal/core/model/adapters.rs

```diff
@@ -3,6 +3,8 @@
 
 //! This module contains adapter models.
 
+use core::{borrow::Borrow, marker::PhantomData};
```

Comment:
Without it there is an issue with `borrow` on type `M`

> no method named `borrow` found for type parameter `M` in the current scope
items from traits can only be used if the trait is in scope

Reply:
Ah yes. This is now it looks like:

```rust
struct SortModelInner<M>
where
    M: Model + 'static,
{
    wrapped_model: M,
    sort_helper: Box<dyn SortHelper<M::Data>>,
    // This vector saves the indices of the elements in sorted order.
    mapping: RefCell<Vec<usize>>,
    notify: ModelNotify,
}
```

and then later there's this code:

```rust
        let changed_data = self.wrapped_model.borrow().row_data(row).unwrap();
```

However, the call to `borrow()` is not needed since `wrapped_model` is stored by value. So you can remove the borrow `use` and the `borrow()` calls and it will compile :)

Reply:
Thanks :-). Done